### PR TITLE
vmselect/prometeus: Add limit param to api/v1/series api endpoint

### PR DESCRIPTION
**Pull request by dmitryk-dk**
Add ```limit``` request param to ```api/v1/series``` which will limiting maximum number of time series, which can be returned. If this limit will be more than ```-search.maxSeries``` flag user get an error. If limit lower or equal to ```0``` system will use value from ```-search.maxSeries``` flag. 

Related issue: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2841

Signed-off-by: dmitryk-dk [d.kozlov@victoriametrics.com](mailto:d.kozlov@victoriametrics.com)

### CODE REVIEW

File: app/vmselect/prometheus/prometheus.go [hunk 1]
```diff
@@ -609,7 +609,14 @@ func SeriesHandler(qt *querytracer.Tracer, startTime time.Time, w http.ResponseW
 	if cp.start == 0 {
 		cp.start = cp.end - defaultStep
 	}
-	sq := storage.NewSearchQuery(cp.start, cp.end, cp.filterss, *maxSeriesLimit)
+	maxSeriesLimit := *maxSeriesLimit
+	if cp.limit > maxSeriesLimit {
+		return fmt.Errorf("limit can't be more than -search.maxSeries flag")
+	}
+	if cp.limit <= 0 {
```
Comment (f41gh7) [inline app/vmselect/prometheus/prometheus.go:616]: It seems, that this check isn't needed.

File: app/vmselect/prometheus/prometheus.go [hunk 2]
```diff
@@ -1022,6 +1032,7 @@ type commonParams struct {
 	end              int64
 	currentTimestamp int64
 	filterss         [][]storage.TagFilter
+	limit            int
```
Comment (f41gh7) [inline app/vmselect/prometheus/prometheus.go:1036]: How limit applied to other API requests? Why it's at common params?

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,7 @@ The following tip changes can be tested by building VictoriaMetrics components f
...
 * FEATURE: [vmagent](https://docs.victoriametrics.com/vmagent.html): add ability to specify additional HTTP headers to send to scrape targets via `headers` section in `scrape_configs`. This can be used when the scrape target requires custom authorization and authentication like in [this stackoverflow question](https://stackoverflow.com/questions/66032498/prometheus-scrape-metric-with-custom-header). For example, the following config instructs sending `My-Auth: top-secret` and `TenantID: FooBar` headers with each request to `http://host123:8080/metrics`:
+* FEATURE: add ability to specify additional request param `limit` to `api/v1/series` endpoint. This can be used if you want to limit the number of series from this api and flag `-search.maxSeries` returns more data than needed.
```
Comment (f41gh7) [inline docs/CHANGELOG.md:29]: Can you also add documentation how and when to use it to the main docs file?

File: app/vmselect/prometheus/prometheus.go [hunk 4]
```diff
@@ -609,7 +609,14 @@ func SeriesHandler(qt *querytracer.Tracer, startTime time.Time, w http.ResponseW
 	if cp.start == 0 {
 		cp.start = cp.end - defaultStep
 	}
-	sq := storage.NewSearchQuery(cp.start, cp.end, cp.filterss, *maxSeriesLimit)
+	maxSeriesLimit := *maxSeriesLimit
+	if cp.limit > maxSeriesLimit {
```
Comment (f41gh7) [inline app/vmselect/prometheus/prometheus.go:613]: Its seems, that this check also isn't needed.

File: docs/CHANGELOG.md (hunk 3 above)
Comment (f41gh7) [reply docs/CHANGELOG.md:29]: Also, commit message doesn't have any reference to the parent issue. It must include it as link to the github page (e.g. https://github.com/VictoriaMetrics/VictoriaMetrics/issues/NUMBER)

**Review by f41gh7 — CHANGES_REQUESTED** [review]

File: app/vmselect/prometheus/prometheus.go (hunk 1 above)
Comment (dmitryk-dk) [reply app/vmselect/prometheus/prometheus.go]: removed

File: app/vmselect/prometheus/prometheus.go (hunk 2 above)
Comment (dmitryk-dk) [reply app/vmselect/prometheus/prometheus.go]: removed from common params

File: docs/CHANGELOG.md (hunk 3 above)
Comment (dmitryk-dk) [reply docs/CHANGELOG.md]: added link to feature request, also updated documentation

File: app/vmselect/prometheus/prometheus.go (hunk 4 above)
Comment (dmitryk-dk) [reply app/vmselect/prometheus/prometheus.go]: logic updated

**Review by f41gh7 — APPROVED** [review]
LGTM

**Review by valyala — APPROVED** [review]

**Pull request merged**